### DISCUSSION TITLE
How to get Loader data in action

### DISCUSSION THREAD

**Discussion by discoverlance-com** [remix-run/remix#1960] [Q&A] [answered]
I am trying to get the loader data in my action and at the moment I can only get it by sending it as a hidden field through the form.

This of course means that users or developers could basically use the browser dev tools to edit and change that data making the data from the hidden inputs in the form unreliable as I am not able to tell whether it's been modified or it's the exact data from the loader.

Is there a way to get the loader data in the action securely without having to use for instance the method I used here or is there any other better way I could pass the data from the loader to the action without the above mentioned security exploit.

**Comment by sergiodxa** [comment, marked as answer]
Imagine you have an API endpoint for POST, and you need to send data, if you need to send data from a previous GET request to another API endpoint you would have to send it as part of the request body to the POST endpoint.

Actions are the same, they are independent API endpoints for POST, if you need data there you either send it in the body (with hidden inputs) or you get it inside the action by running any DB query or fetch you do in the loader also in the action.

**Reply by discoverlance-com** [reply]
Ok I do understand your point. I just thought there could be a way to do that as fetching data in the action when it's not really data from the form the user has entered could result in a delay that's not expected (should the api fail for some reason, if it were to fail from the loader then I could deal with that in an error boundary directly without allowing the user to try to enter the other data required in the form in the first place).

But looking at the purpose of actions now as you have mentioned, I think that either way with hidden inputs or fetching the query in the actions is the expected behaviour of the action function so will stick with any for now. Thanks.
